sway/handlers.c

```diff
@@ -874,17 +874,28 @@ static bool handle_pointer_button(wlc_handle view, uint32_t time, const struct w
 bool handle_pointer_scroll(wlc_handle view, uint32_t time, const struct wlc_modifiers* modifiers,
 		uint8_t axis_bits, double _amount[2]) {
 	if (!(modifiers->mods ^ config->floating_mod)) {
+		bool handled = false;
 		int x_amount = (int)_amount[0];
 		int y_amount = (int)_amount[1];
-		if (x_amount > 0) {
+
+		if (x_amount > 0 && strcmp(config->floating_scroll_up_cmd, "")) {
 			handle_command(config->floating_scroll_up_cmd);
-		} else if (x_amount < 0) {
+			handled = true;
+		} else if (x_amount < 0 && strcmp(config->floating_scroll_down_cmd, "")) {
 			handle_command(config->floating_scroll_down_cmd);
+			handled = true;
 		}
-		if (y_amount > 0) {
+
+		if (y_amount > 0 && strcmp(config->floating_scroll_right_cmd, "")) {
 			handle_command(config->floating_scroll_right_cmd);
-		} else if (y_amount < 0) {
+			handled = true;
+		} else if (y_amount < 0 && strcmp(config->floating_scroll_left_cmd, "")) {
 			handle_command(config->floating_scroll_left_cmd);
+			handled = true;
+		}
+
+		if (handled) {
+			return EVENT_HANDLED;
```

Comment:
I wasn't sure how sideway scrolling would register (if both vertical and horizontal components would be != 0).